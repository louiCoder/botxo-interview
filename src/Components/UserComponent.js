import React from 'react';
import './styles.scss';

export default function UserComponent ({ changeName, changeAge }) {
  return (
    <div className="users-container">
      <div className="name-and-age">
        <div>
          <p>Name</p>
          <input type="text" onChange={(e) => changeName(e)} className="input-field" />
        </div>
        <div>
          <p>Age</p>
          <input type="number" onChange={(e) => changeAge(e)} className="input-age" />
        </div>
      </div>
      {/* <div>
        <img src={Trash} alt="delete_icon" onClick={() => alert()} />
        <svg style={{ width: '24px', height: '24px' }} viewBox="0 0 24 24" onClick={() => alert()}>
          <path
            fill="currentColor"
            d="M19,4H15.5L14.5,3H9.5L8.5,4H5V6H19M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19Z"
          />
        </svg>
      </div> */}
    </div>
  );
}
