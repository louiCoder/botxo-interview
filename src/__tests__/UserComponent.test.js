import React from 'react';
import { shallow } from 'enzyme';
import UserComponent from '../Components/UserComponent';

describe('<UserComponent /> component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      changeName: jest.fn(),
      changeAge: jest.fn(),
      addToGenericList: jest.fn(),
      name: 'louis',
      age: 10
    };
    wrapper = shallow(<UserComponent {...props} onChange={() => jest.fn()} />);
  });

  it('Render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('#name-input').prop('value')).toEqual('louis');
    expect(wrapper.find('#age-input').prop('value')).toEqual(10);
  });

  it('calls the changeName function with an object', () => {
    wrapper.find('#name-input').simulate('change', { target: { value: 'louis' } });
    expect(props.changeName).toBeCalledWith({ target: { value: 'louis' } });
  });

  it('switches from one view to another', () => {
    wrapper.find('#age-input').simulate('change', { target: { value: 20 } });
    expect(props.changeAge).toBeCalledWith({ target: { value: 20 } });
  });

  it("calls changeName function when there's a change in the name ", () => {
    wrapper.find('#name-input').simulate('change', { target: { value: 'michael' } });
    expect(props.changeName).toBeCalled();
    expect(props.changeName).toBeCalledWith({ target: { value: 'michael' } });
  });

  it("should call changeAge function when there's a change in the age ", () => {
    wrapper.find('#age-input').simulate('change', { target: { value: 20 } });
    expect(props.changeAge).toBeCalled();
    expect(props.changeAge).toBeCalledWith({ target: { value: 20 } });
  });
});
