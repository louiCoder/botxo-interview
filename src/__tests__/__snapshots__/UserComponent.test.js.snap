// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component Render without crashing 1`] = `ShallowWrapper {}`;

exports[`<UserComponent /> component Render without crashing 1`] = `ShallowWrapper {}`;
